tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs

```diff
@@ -0,0 +1,119 @@
+                Console.WriteLine($"Driver={Driver}, Scheme={Scheme}, ButtonPin={ButtonPin}, PressedValue={PressedValue}, OnValue={OnValue}");
+            }
+
+            using (GpioController gpio = CreateGpioController())
+            {
+                using (var cancelEvent = new ManualResetEvent(false))
+                {
+                    Console.WriteLine($"Listening for button presses on GPIO {Enum.GetName(typeof(PinNumberingScheme), Scheme)} pin {ButtonPin} . . .");
+
+                    // This example runs until Ctrl+C (or Ctrl+Break) is pressed, so register a local function handler.
+                    Console.CancelKeyPress += Console_CancelKeyPress;
+                    gpio.OpenPin(ButtonPin);
+
+                    // Set the mode based on if input pull-up resistors are supported.
+                    PinMode inputMode = gpio.IsPinModeSupported(ButtonPin, PinMode.InputPullUp) ? PinMode.InputPullUp : PinMode.Input;
+                    gpio.SetPinMode(ButtonPin, inputMode);
+
+                    // Open the GPIO pin connected to the LED if one was specified.
+                    if (LedPin != null)
+                    {
+                        gpio.OpenPin(LedPin.Value, PinMode.Output);
+                        gpio.Write(LedPin.Value, OffValue);
+                    }
+
+                    // Declare a local function to handle the pin value changed events.
+                    int count = 0;
+                    void valueChangeHandler(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
+                    {
+                        if (LedPin != null)
+                        {
+                            PinValue ledValue = pinValueChangedEventArgs.ChangeType == PressedValue ? OnValue : OffValue;
+                            gpio.Write(LedPin.Value, ledValue);
+                        }
+
+                        var pressedOrReleased = pinValueChangedEventArgs.ChangeType == PressedValue ? "pressed" : "released";
+                        Console.WriteLine($"[{count++}] Button {pressedOrReleased}: logicalPinNumber={pinValueChangedEventArgs.PinNumber}, ChangeType={pinValueChangedEventArgs.ChangeType}");
+                    }
+
+                    // Set the event handler for changes to the pin value.
+                    PinEventTypes bothPinEventTypes = PinEventTypes.Falling | PinEventTypes.Rising;
+                    gpio.RegisterCallbackForPinValueChangedEvent(ButtonPin, bothPinEventTypes, valueChangeHandler);
+
+                    // Wait for the cancel (Ctrl+C) console event.
+                    cancelEvent.WaitOne();
+
+                    gpio.ClosePin(ButtonPin);
+                    if (LedPin != null)
+                    {
+                        gpio.ClosePin(LedPin.Value);
+                    }
+
+                    // Unregister the event handler for changes to the pin value
+                    gpio.UnregisterCallbackForPinValueChangedEvent(ButtonPin, valueChangeHandler);
```

Comment:
This test could be more demonstrative by covering/using more than one event. For example this is my event handler test I used, it is not perfect but covering more test scenarios, it uses 2 buttons though, just to give you some other idea:
```csharp
ManualResetEvent mre = new ManualResetEvent(false);
                using (GpioController controller = new GpioController(PinNumberingScheme.Logical , UnixDriver.InitUnixDriver()))
                {
                    Console.WriteLine("event registration test: " );
                    PinValue ledValue = PinValue.Low;
                    controller.OpenPin(23, PinMode.Output);
                    controller.OpenPin(18, PinMode.Input);
                    controller.OpenPin(25, PinMode.Input);
                    int clickCount = 0;
                    controller.RegisterCallbackForPinValueChangedEvent(18, PinEventTypes.Rising, (o, e) => {
                        Console.WriteLine("Raising event 1 called");
                        ledValue = toggleValue(ledValue);
                        controller.Write(23, ledValue);
                        clickCount ++;
                            if (clickCount > 10) {
                                mre.Set();
                            }
                        });  
                    PinChangeEventHandler callback = (o, e) => { 
                        Console.WriteLine("Raising event 2 delegate for remove called");
                    };
                    controller.RegisterCallbackForPinValueChangedEvent(18, PinEventTypes.Rising, callback);
                    controller.RegisterCallbackForPinValueChangedEvent(18, PinEventTypes.Rising, (o, e) => {
                        Console.WriteLine("Raising event 3 called");
                    });

                    controller.RegisterCallbackForPinValueChangedEvent(18, PinEventTypes.Falling, (o, e) => {
                        Console.WriteLine("Falling event 1 called");
                    });
                    controller.RegisterCallbackForPinValueChangedEvent(25, PinEventTypes.Rising, (o, e) => {
                        Console.WriteLine("Remove callback event occured");
                        controller.UnregisterCallbackForPinValueChangedEvent(18, callback);
                    });
                    mre.WaitOne();
                }
```

Reply:
Thanks for the feedback @buyaa-n.  I have not studied the events yet as that is next up on part of my Mcp23xxx binding exercise.  Let's keep as is for now and maybe circle back if needed.

Reply:
Sounds good, thanks